test/index.js

```diff
@@ -522,6 +533,86 @@ test('equals', function() {
+  eq(Z.lte(new Boolean(true), new Boolean(true)), true);
+  eq(Z.lte(false, new Boolean(false)), false);
+  eq(Z.lte(new Boolean(false), false), false);
+  eq(Z.lte(true, new Boolean(true)), false);
+  eq(Z.lte(new Boolean(true), true), false);
+  eq(Z.lte(0, 0), true);
+  eq(Z.lte(0, -0), true);
+  eq(Z.lte(-0, 0), true);
+  eq(Z.lte(-0, -0), true);
+  eq(Z.lte(NaN, NaN), true);
+  eq(Z.lte(Infinity, Infinity), true);
+  eq(Z.lte(Infinity, -Infinity), false);
+  eq(Z.lte(-Infinity, Infinity), true);
+  eq(Z.lte(-Infinity, -Infinity), true);
+  eq(Z.lte(NaN, Math.PI), false);
+  eq(Z.lte(Math.PI, NaN), false);
+  eq(Z.lte(new Number(0), new Number(0)), true);
+  eq(Z.lte(new Number(0), new Number(-0)), true);
+  eq(Z.lte(new Number(-0), new Number(0)), true);
+  eq(Z.lte(new Number(-0), new Number(-0)), true);
+  eq(Z.lte(new Number(NaN), new Number(NaN)), true);
+  eq(Z.lte(new Number(Infinity), new Number(Infinity)), true);
+  eq(Z.lte(new Number(Infinity), new Number(-Infinity)), false);
+  eq(Z.lte(new Number(-Infinity), new Number(Infinity)), true);
+  eq(Z.lte(new Number(-Infinity), new Number(-Infinity)), true);
+  eq(Z.lte(new Number(NaN), new Number(Math.PI)), false);
+  eq(Z.lte(new Number(Math.PI), new Number(NaN)), false);
+  eq(Z.lte(42, new Number(42)), true);
+  eq(Z.lte(new Number(42), 42), true);
+  eq(Z.lte(new Date(0), new Date(0)), true);
+  eq(Z.lte(new Date(0), new Date(1)), true);
+  eq(Z.lte(new Date(1), new Date(0)), false);
+  eq(Z.lte(new Date(1), new Date(1)), true);
+  eq(Z.lte(new Date(NaN), new Date(NaN)), true);
+  eq(Z.lte('', ''), true);
+  eq(Z.lte('abc', 'abc'), true);
+  eq(Z.lte('abc', 'xyz'), true);
+  eq(Z.lte('xyz', 'abc'), false);
+  eq(Z.lte(new String(''), new String('')), true);
+  eq(Z.lte(new String('abc'), new String('abc')), true);
+  eq(Z.lte(new String('abc'), new String('xyz')), true);
+  eq(Z.lte(new String('xyz'), new String('abc')), false);
+  eq(Z.lte('abc', new String('abc')), false);
+  eq(Z.lte(new String('abc'), 'abc'), false);
+  eq(Z.lte([], []), true);
+  eq(Z.lte([1, 2], [1, 2]), true);
+  eq(Z.lte([1, 2, 3], [1, 2]), false);
+  eq(Z.lte([1, 2], [1, 2, 3]), true);
+  eq(Z.lte([1, 2], [2, 1]), true);
+  eq(Z.lte([0], [-0]), true);
+  eq(Z.lte([NaN], [NaN]), true);
+  eq(Z.lte(ones, ones), true);
+  eq(Z.lte(args(), args()), true);
+  eq(Z.lte(args(1, 2), args(1, 2)), true);
+  eq(Z.lte(args(1, 2, 3), args(1, 2)), false);
+  eq(Z.lte(args(1, 2), args(1, 2, 3)), true);
+  eq(Z.lte(args(1, 2), args(2, 1)), true);
+  eq(Z.lte(args(0), args(-0)), true);
+  eq(Z.lte(args(NaN), args(NaN)), true);
+  eq(Z.lte(Identity(Identity(Identity(0))), Identity(Identity(Identity(0)))), true);
+  eq(Z.lte(Identity(Identity(Identity(0))), Identity(Identity(Identity(1)))), true);
+  eq(Z.lte(Identity(Identity(Identity(1))), Identity(Identity(Identity(0)))), false);
+  eq(Z.lte(Identity(Identity(Identity(1))), Identity(Identity(Identity(1)))), true);
+});
```

Comment:
Given your implementation of `Array$prototype$lte`, it may not be needed. I'm having trouble coming up with a test case that wouldn't be covered by the handling of circular structures in `equals`.